### PR TITLE
8.0 branch

### PR DESCRIPTION
**Pull request by mileo**
Hello,

A few months ago I created a "PR" for the creation of a branch 8.0 for this repository. 

> Can someone that have push permission to upstream, create the 8.0 branch? https://github.com/OCA/l10n-brazil/pull/109

Now this is a pull request of months of work from @danimaribeiro @mileo @mstuttgart @Mackilem to migrate l10n_br to 8.0. Here is the currenty WIP:
1. Make all modules installable; OK
2. Fix bugs and make then operational; OK
3. Migrate to new API and flake8 pass:
- [X] l10n_br_account
- [ ] l10n_br_account_payment
- [x] l10n_br_account_product
- [ ] l10n_br_account_product_service
- [ ] l10n_br_account_service
- [ ] l10n_br_account_voucher
- [x] l10n_br_base
- [x] l10n_br_crm
- [ ] l10n_br_crm_zip
- [x] l10n_br_data_account
- [x] l10n_br_data_account_product
- [ ] l10n_br_data_account_service
- [x] l10n_br_data_base
- [x] l10n_br_delivery
- [ ] l10n_br_hr_timesheet_invoice
- [x] l10n_br_product
- [x] l10n_br_purchase
- [x] l10n_br_sale
- [x] l10n_br_sale_product
- [ ] l10n_br_sale_service
- [x] l10n_br_sale_stock
- [x] l10n_br_stock
- [ ] l10n_br_zip

During migration we also did everything possible to improve functionality. There have been improvements in crm ( company and contact creating on lead ) and account ( full operational MD-E ). If anyone wants to help or is already working on migration we will be happy to know.

In https://github.com/OCA/l10n-brazil/pull/109#issuecomment-64740403 and as told before yesterday on skype (with @danimaribeiro): @renatonlima informed us that he had worked on migration in November and publish your code today. For some unknown reason the branch 8.0 has not already been created and we had no feed back. So that's the PR of our work.

But both KMEE as other contributors are waiting a long time to be able to contribute more actively and without a very large delay. We would like to request that someone looked fondly our actions so that we can continue to contribute more and better. Thus, we would not have only one company handling the official repository of the Brazilian location. We can contribute to reviewing and analyzing third-party code task to be able to streamline the approvals in our location.

The intention of this request is to contribute to the development of new features and fixes in odoo environment. In KMEE we always seek to promote what we are working from the first commit. During this one year that we are official partner, there was no lack of good intention, and let's say that it was not the easiest year to get into the game and we hadn't any malfeasance.

Maybe it's my point of view, to want things to evolve too fast, anyway, I'm sorry if there was any misunderstanding

 Best regards

### CODE REVIEW

**Comment by danimaribeiro** [comment]
Any news on the migration guys.
My first email on the Odoo-Brasil group talking about the 8.0 version was in october/2014.
It has been exactly 4 months waiting, in the meantime I started and stopped working in the migration a few times, because the emails saying the migration was coming.
I hope someone publish the code already migrated or accept our contribution, then we can move forward on this.

If we are not transparent in our actions, and how we deal with contributions, it's hard to expect contributions.

**Comment by gurneyalex** [comment]
@mileo  I added you github login information in the OCA database

No CLA found for @Mackilem or @danimaribeiro

**Comment by mileo** [comment]
@gurneyalex Thanks. I Contacted @danimaribeiro and @Mackilem to sign OCA CLA.

[]s

**Comment by mileo** [comment]
up!

**Comment by shingonoide** [comment]
The test script seem not working ok.

> $ ./.travis_test_script.sh
> /home/travis/build/OCA/openerp
> fiscal_rules  l10n-brazil  odoo_8.0.latest.tar.gz  openerp  server-env-tools
> /home/travis/virtualenv/python2.7_with_system_site_packages/local/lib/python2.7/site-packages/pkg_resources/**init**.py:1222: UserWarning: /home/travis/.python-eggs is writable by group/others and vulnerable to attack when used with get_resource_filename. Consider a more secure location (set with .set_extraction_path or the PYTHON_EGG_CACHE environment variable).
>   warnings.warn(msg, UserWarning)
> Usage: openerp-server [options]
> openerp-server: error: option --addons-path: no such directory: '../l10n_br_core'
> The command "./.travis_test_script.sh" exited with 0.
> Done. Your build exited with 0.

**Comment by danimaribeiro** [comment]
@shingonoide Yes, it's not working and also has some unit tests not passing.
Maybe would be better to change from travis to OCA runbot, I'll check on that.

**Comment by mileo** [comment]
In the KMEE we recognize the investment Akretion and always try to be correct when we send a pull request or suggestion.

In late 2014 https://github.com/OCA/l10n-brazil/pull/109 request the creation of the Branch 8.0 but the pull request was closed by @renatonlima. Surely the best way to realize the pull request was module by module. But with the creation of the empty branch not allowed, it is impossible...

So the solution was to carry out the migration process in a fork https://github.com/odoo-brazil/l10n-brazil and perform a pull request from the 8.0 branch only to record our work.

Yesterday I spoke with the contributors pull request https://github.com/OCA/l10n-brazil/pull/124 and anyone was pleased with these news PR of Akretion. It looks like that our work has no importance because of this tyranny.

If the location is in the OCA arms, great, guarantees more security to customers, but our pull request has been done before and should be performed before the merge. We will be happy to correct any potential problems and if necessary perform pull separate requests.

I'm sure also that since we started to contribute in late 2011 we make mistakes and we are proud to have learned from them.

I hope that the OCA can evaluate our position in order to prevent people stop contributing to the official branch.

Best Regards

File: l10n_br_account/res_partner.py [hunk 1]
```diff
@@ -17,43 +17,40 @@
...
+    type = fields.Selection([('input', 'Entrada'), ('output', 'Saida')],
+                             string='Tipo')
+    type_tax_use = fields.Selection(
+        [('sale', 'Sale'), ('purchase', 'Purchase'), ('all', 'All')],
+        string='Tax Application')
+    inv_copy_note = fields.Boolean(string=u'Copiar Observação na Nota Fiscal')
+    asset_operation = fields.Boolean(string=u'Operação de Aquisição de Ativo',
+        help=u"""Caso seja marcada essa opção, será incluido o IPI na base de
+            calculo do ICMS.""")
+    id_dest = fields.Selection([('1', u'Operação interna'),
```
Comment (renatonlima) [inline l10n_br_account/res_partner.py:42]: This filed id_dest should be in l10n_br_account.product.cfop object, I did this in https://github.com/akretion/l10n-brazil/blob/migration/l10n_br_account_product/l10n_br_account_product/models/l10n_br_account_product.py#L53

**Comment by rvalyi** [comment]
superseded by ongoing PR

**Pull request closed without merging**